sycl/test/regression/check_config_processing.cpp

```diff
@@ -0,0 +1,158 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+//
+// RUN: env TEST=SPACE_AT_FIRST_POSITION %t.out > %t1.conf
+// RUN: env TEST=EXCESS_SPACE env SYCL_CONFIG_FILE_NAME=%t1.conf %t.out
+//
+// RUN: env TEST=SPACE_AT_LAST_POSITION %t.out > %t2.conf
+// RUN: env TEST=EXCESS_SPACE env SYCL_CONFIG_FILE_NAME=%t2.conf %t.out
+//
+// RUN: env TEST=SPACE_BEFORE_ASSIGNMENT %t.out > %t3.conf
+// RUN: env TEST=EXCESS_SPACE env SYCL_CONFIG_FILE_NAME=%t3.conf %t.out
+//
+// RUN: env TEST=SPACE_AFTER_ASSIGNMENT %t.out > %t4.conf
+// RUN: env TEST=EXCESS_SPACE env SYCL_CONFIG_FILE_NAME=%t4.conf %t.out
+//
+// RUN: env TEST=VARIABLE_NAME_BIGGER_THAN_MAX_CONFIG_NAME %t.out > %t5.conf
+// RUN: env TEST=INCORRECT_VARIABLE_NAME env SYCL_CONFIG_FILE_NAME=%t5.conf %t.out
+//
+// RUN: env TEST=VARIABLE_WITHOUT_NAME %t.out > %t6.conf
+// RUN: env TEST=INCORRECT_VARIABLE_NAME env SYCL_CONFIG_FILE_NAME=%t6.conf %t.out
+//
+// RUN: env TEST=VARIABLE_VALUE_BIGGER_THAN_MAX_CONFIG_VALUE %t.out > %t7.conf
+// RUN: env TEST=INCORRECT_VARIABLE_VALUE env SYCL_CONFIG_FILE_NAME=%t7.conf %t.out
+//
+// RUN: env TEST=VARIABLE_WITHOUT_VALUE %t.out > %t8.conf
+// RUN: env TEST=INCORRECT_VARIABLE_VALUE env SYCL_CONFIG_FILE_NAME=%t8.conf %t.out
+//
+// RUN: env TEST=COMPLEX_CONFIG %t.out > %t9.conf
+// RUN: env TEST=INCORRECT_VARIABLE_VALUE env SYCL_CONFIG_FILE_NAME=%t9.conf %t.out
+//
+// RUN: env TEST=SET_CORRECT_ENVIRONMENT %t.out > %t10.conf
+// RUN: env TEST=CORRECT_CONFIG_FILE env SYCL_CONFIG_FILE_NAME=%t10.conf %t.out
+// RUN: ls | grep dot
+// RUN: rm *.dot
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <regex>
+
+int main() {
+  std::string testEnvVarValue;
+  if (getenv("TEST")) {
+    testEnvVarValue = getenv("TEST");
+  } else {
+    throw sycl::exception(sycl::make_error_code(sycl::errc::runtime),
+                          "Environment is not found");
+  }
+
+  // Сonfig creation
```

Comment:
Are there any checks on line 49 to 108?

Reply:
No, on these lines we only write the config based on the TEST environment variable.

Reply:
these look slike debug info and it can be removed from the final version of the test.

Reply:
We check more than three negative cases (nine, to be exact). Therefore, if we will create an individual config file to test every possible case, we will have too many excess files. Yes, this will shorten test code, but we will significantly expand the directory with this test.

Reply:
The test configs can be moved to a separate directory. From my perspective, it is easier to read the file rather than the code that generates it.
but this is NIT request. So feel free to leave it as is.